### DISCUSSION TITLE
Is there a better way to set `index` as a "permanent" dim rather than a `MultiIndex` dim?

### DISCUSSION THREAD

**Discussion by ahuang11** [pydata/xarray#4800] [Q&A] [answered]
```
import xarray as xr
import numpy as np
da = xr.tutorial.open_dataset('air_temperature')['air'].isel(time=0)
da[0, 0] = np.nan
da[15, 0] = np.nan

da = da.stack({'index': ('lat', 'lon')}).dropna('index')
da = da.reset_index('index').assign_coords(**{'index': range(len(da['index']))})
da
```
I want:
![image](https://user-images.githubusercontent.com/15331990/104369635-5d9fdf80-54e3-11eb-8333-d94d740d9761.png)

To become this:
![image](https://user-images.githubusercontent.com/15331990/104369673-6f818280-54e3-11eb-9b2a-b854e417da58.png)

**Comment by keewis** [comment, marked as answer]
not sure if there's a better way to do this, but you can combine the expressions and use `da.sizes["index"]` instead of `len(da["index"])`:
```python
da = (
    da.stack({"index": ("lat", "lon")})
    .dropna("index")
    .reset_index("index")
    .assign_coords({"index": lambda x: np.arange(x.sizes["index"])})
)
```

**Reply by ahuang11** [reply]
Thanks! Didn't know about `sizes`!
